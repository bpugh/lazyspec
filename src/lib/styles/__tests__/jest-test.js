const fs = require('fs');

const describeModule = require('../../describeModule');
const specUnit = require('../jest');

function expectSpec(moduleName) {
  const src = fs.readFileSync(`spec/fixtures/${moduleName}.js`).toString();
  const description = describeModule(moduleName, src);

  const spec = specUnit(description);

  expect(spec).toMatchSnapshot();
}

describe('describeModule', () => {
  it('should describe simple exports', () => {
    expectSpec('esExports');
  });

  it('should describe a createClass component', () => {
    expectSpec('CreateClass');
  });

  it('should describe an ES6 component', () => {
    expectSpec('SafeComponent');
  });

  it('should describe an unsafe ES6 component', () => {
    expectSpec('UnsafeComponent');
  });

  it('should describe an ES6 with Flow component', () => {
    expectSpec('FlowComponent');
  });

  it('should describe a stateless function component', () => {
    expectSpec('StatelessFunction');
  });
});
