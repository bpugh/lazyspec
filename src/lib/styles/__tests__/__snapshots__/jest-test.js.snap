exports[`describeModule should describe a createClass component 1`] = `
"import React from \'react\';
import renderer from \'react-test-renderer\';

describe(\'CreateClass\', () => {
  it(\'exists\', () => {
    expect(CreateClass).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <CreateClass  />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});"
`;

exports[`describeModule should describe a stateless function component 1`] = `
"import React from \'react\';
import renderer from \'react-test-renderer\';

describe(\'StatelessFunction\', () => {
  it(\'exists\', () => {
    expect(StatelessFunction).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <StatelessFunction  />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});"
`;

exports[`describeModule should describe an ES6 component 1`] = `
"import React from \'react\';
import renderer from \'react-test-renderer\';

describe(\'SafeComponent\', () => {
  it(\'exists\', () => {
    expect(SafeComponent).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <SafeComponent  />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});"
`;

exports[`describeModule should describe an ES6 with Flow component 1`] = `
"import React from \'react\';
import renderer from \'react-test-renderer\';

describe(\'FlowComponent\', () => {
  it(\'exists\', () => {
    expect(FlowComponent).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <FlowComponent label=\"mockstring\" />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});"
`;

exports[`describeModule should describe an unsafe ES6 component 1`] = `
"import React from \'react\';
import renderer from \'react-test-renderer\';

describe(\'UnsafeComponent\', () => {
  it(\'exists\', () => {
    expect(UnsafeComponent).toBeTruthy();
  });

  xit(\'renders\', () => {
    const comp = <UnsafeComponent foo=\"TODO mock\" />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});"
`;

exports[`describeModule should describe simple exports 1`] = `
"describe(\'esExports\', () => {
  it(\'exists\', () => {
    expect(esExports).toBeTruthy();
  });
});
describe(\'refConst\', () => {
  it(\'exists\', () => {
    expect(refConst).toBeTruthy();
  });
});
describe(\'refFunc\', () => {
  it(\'exists\', () => {
    expect(refFunc).toBeTruthy();
  });
});
describe(\'inlineConst\', () => {
  it(\'exists\', () => {
    expect(inlineConst).toBeTruthy();
  });
});
describe(\'inlineFunc\', () => {
  it(\'exists\', () => {
    expect(inlineFunc).toBeTruthy();
  });
});"
`;
